dauthz/core.py

```diff
@@ -1,13 +1,46 @@
+# from casbin_adapter.adapter import Adapter as CasbinAdapter
+import logging
+
 from casbin import Enforcer
+from django.conf import settings
+
+from .utils import import_class
 
 
 class ProxyEnforcer(Enforcer):
-    def __init__(self):
-        # todo read configuration, initialize Enforcer...
-        model = None
-        adapter = None 
-        super.__init__(model, adapter)
-        pass
+    name = None
+    logger = None
+    enable_log = None
+
+    def __init__(self, enforcer_name: str, enforcer_conf: dict):
+        self.name = enforcer_name
+        self.logger = logging.getLogger(self.name)
+
+        model_type = enforcer_conf["MODEL"]["CONFIG_TYPE"]
+        if model_type == "file":
+            model = enforcer_conf["MODEL"]["CONFIG_FILE_PATH"]
+        else:
+            model = enforcer_conf["MODEL"]["CONFIG_TEXT"]
+        model = str(model)
+        self.enable_log = enforcer_conf["LOG"]["ENABLED"]
+
+        adapter_loc = enforcer_conf["ADAPTER"]["NAME"]
+        adapter_class = import_class(adapter_loc)
+        adapter = adapter_class()
+        super().__init__(model, adapter)
+
+        watcher = enforcer_conf.get("WATCHER")
+        if watcher:
+            self.set_watcher(watcher)
+        role_manager = enforcer_conf.get("ROLE_MANAGER")
+        if role_manager:
+            self.set_role_manager(role_manager)
+        self.logger.debug("Casbin enforcer initialised")
 
 
-enforcer = ProxyEnforcer()
+enforcers = {}
```

Comment:
I think this provides an entry for external calls like this:

```python
dauthz.core.enforcer.enforce(...)
dauthz.core.enforcer.get_policy(...)
...
```

Reply:
changed. add default enforcer.
![image](https://user-images.githubusercontent.com/75596353/184564056-7f99696a-1ff3-448b-af5c-9d22a5ec9729.png)
set the default enforcer in the conf file like laravel-authz